zerver/actions/reminders.py

```diff
@@ -0,0 +1,40 @@
+import datetime
+
+from zerver.actions.message_send import check_message
+from zerver.actions.scheduled_messages import do_schedule_messages
+from zerver.lib.addressee import Addressee
+from zerver.lib.reminders import access_message_for_reminder
+from zerver.models import Client, ScheduledMessage, UserProfile
+
+
+def schedule_reminder_for_message(
+    current_user: UserProfile,
+    client: Client,
+    message_id: int,
+    deliver_at: datetime.datetime,
+) -> int:
+    message = access_message_for_reminder(current_user, message_id)
+    # Even though reminder will be sent from NOTIFICATION_BOT, we still
+    # set current_user as the sender here to help us make the permission checks easier.
+    addressee = Addressee.for_user_profile(current_user)
+    # This assumes that any logged in user can send a message to self.
+    send_request = check_message(
+        current_user,
+        client,
+        addressee,
+        # Store the content for displaying scheduled reminders in the UI,
+        # it will be modified before sending to the user.
+        message.content,
```

Comment:
This is just the original message content? I kinda feel like we might want `get_reminder_formatted_content`.

Reply:
Talked to Alya and I think that we indeed want `get_reminder_formatted_content`. 

If we store that, then we can have a nicer fallback if you can't access the target message anymore (say, because it's been deleted or is no longer accessible) -- just send the originally stored reminder text. (But we'll still use the latest location for the message if accessible).

Reply:
Changed to use get_reminder_formatted_content